File: SDKs/googleapis.com/vision/v1p1beta1/python/src/sdk/models/shared/googlecloudvisionv1p1beta1faceannotation.py
from dataclasses import dataclass, field
from typing import Enum,List,Optional
from dataclasses_json import dataclass_json
from . import googlecloudvisionv1p1beta1boundingpoly
from . import googlecloudvisionv1p1beta1boundingpoly
from . import googlecloudvisionv1p1beta1faceannotationlandmark

class GoogleCloudVisionV1p1beta1FaceAnnotationAngerLikelihoodEnum(str, Enum):
    UNKNOWN = "UNKNOWN"
    VERY_UNLIKELY = "VERY_UNLIKELY"
    UNLIKELY = "UNLIKELY"
    POSSIBLE = "POSSIBLE"
    LIKELY = "LIKELY"
    VERY_LIKELY = "VERY_LIKELY"

class GoogleCloudVisionV1p1beta1FaceAnnotationBlurredLikelihoodEnum(str, Enum):
    UNKNOWN = "UNKNOWN"
    VERY_UNLIKELY = "VERY_UNLIKELY"
    UNLIKELY = "UNLIKELY"
    POSSIBLE = "POSSIBLE"
    LIKELY = "LIKELY"
    VERY_LIKELY = "VERY_LIKELY"

class GoogleCloudVisionV1p1beta1FaceAnnotationHeadwearLikelihoodEnum(str, Enum):
    UNKNOWN = "UNKNOWN"
    VERY_UNLIKELY = "VERY_UNLIKELY"
    UNLIKELY = "UNLIKELY"
    POSSIBLE = "POSSIBLE"
    LIKELY = "LIKELY"
    VERY_LIKELY = "VERY_LIKELY"

class GoogleCloudVisionV1p1beta1FaceAnnotationJoyLikelihoodEnum(str, Enum):
    UNKNOWN = "UNKNOWN"
    VERY_UNLIKELY = "VERY_UNLIKELY"
    UNLIKELY = "UNLIKELY"
    POSSIBLE = "POSSIBLE"
    LIKELY = "LIKELY"
    VERY_LIKELY = "VERY_LIKELY"

class GoogleCloudVisionV1p1beta1FaceAnnotationSorrowLikelihoodEnum(str, Enum):
    UNKNOWN = "UNKNOWN"
    VERY_UNLIKELY = "VERY_UNLIKELY"
    UNLIKELY = "UNLIKELY"
    POSSIBLE = "POSSIBLE"
    LIKELY = "LIKELY"
    VERY_LIKELY = "VERY_LIKELY"

class GoogleCloudVisionV1p1beta1FaceAnnotationSurpriseLikelihoodEnum(str, Enum):
    UNKNOWN = "UNKNOWN"
    VERY_UNLIKELY = "VERY_UNLIKELY"
    UNLIKELY = "UNLIKELY"
    POSSIBLE = "POSSIBLE"
    LIKELY = "LIKELY"
    VERY_LIKELY = "VERY_LIKELY"

class GoogleCloudVisionV1p1beta1FaceAnnotationUnderExposedLikelihoodEnum(str, Enum):
    UNKNOWN = "UNKNOWN"
    VERY_UNLIKELY = "VERY_UNLIKELY"
    UNLIKELY = "UNLIKELY"
    POSSIBLE = "POSSIBLE"
    LIKELY = "LIKELY"
    VERY_LIKELY = "VERY_LIKELY"


@dataclass_json
@dataclass
class GoogleCloudVisionV1p1beta1FaceAnnotation:
    anger_likelihood: Optional[GoogleCloudVisionV1p1beta1FaceAnnotationAngerLikelihoodEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'angerLikelihood' }})
    blurred_likelihood: Optional[GoogleCloudVisionV1p1beta1FaceAnnotationBlurredLikelihoodEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'blurredLikelihood' }})
    bounding_poly: Optional[googlecloudvisionv1p1beta1boundingpoly.GoogleCloudVisionV1p1beta1BoundingPoly] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'boundingPoly' }})
    detection_confidence: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'detectionConfidence' }})
    fd_bounding_poly: Optional[googlecloudvisionv1p1beta1boundingpoly.GoogleCloudVisionV1p1beta1BoundingPoly] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'fdBoundingPoly' }})
    headwear_likelihood: Optional[GoogleCloudVisionV1p1beta1FaceAnnotationHeadwearLikelihoodEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'headwearLikelihood' }})
    joy_likelihood: Optional[GoogleCloudVisionV1p1beta1FaceAnnotationJoyLikelihoodEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'joyLikelihood' }})
    landmarking_confidence: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'landmarkingConfidence' }})
    landmarks: Optional[List[googlecloudvisionv1p1beta1faceannotationlandmark.GoogleCloudVisionV1p1beta1FaceAnnotationLandmark]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'landmarks' }})
    pan_angle: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'panAngle' }})
    roll_angle: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'rollAngle' }})
    sorrow_likelihood: Optional[GoogleCloudVisionV1p1beta1FaceAnnotationSorrowLikelihoodEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'sorrowLikelihood' }})
    surprise_likelihood: Optional[GoogleCloudVisionV1p1beta1FaceAnnotationSurpriseLikelihoodEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'surpriseLikelihood' }})
    tilt_angle: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'tiltAngle' }})
    under_exposed_likelihood: Optional[GoogleCloudVisionV1p1beta1FaceAnnotationUnderExposedLikelihoodEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'underExposedLikelihood' }})
    

File: SDKs/googleapis.com/vision/v1p1beta1/typescript/src/sdk/models/shared/googlecloudvisionv1p1beta1faceannotation.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { GoogleCloudVisionV1p1beta1BoundingPoly } from "./googlecloudvisionv1p1beta1boundingpoly";
import { GoogleCloudVisionV1p1beta1FaceAnnotationLandmark } from "./googlecloudvisionv1p1beta1faceannotationlandmark";


export enum GoogleCloudVisionV1p1beta1FaceAnnotationAngerLikelihoodEnum {
    Unknown = "UNKNOWN",
    VeryUnlikely = "VERY_UNLIKELY",
    Unlikely = "UNLIKELY",
    Possible = "POSSIBLE",
    Likely = "LIKELY",
    VeryLikely = "VERY_LIKELY"
}

export enum GoogleCloudVisionV1p1beta1FaceAnnotationBlurredLikelihoodEnum {
    Unknown = "UNKNOWN",
    VeryUnlikely = "VERY_UNLIKELY",
    Unlikely = "UNLIKELY",
    Possible = "POSSIBLE",
    Likely = "LIKELY",
    VeryLikely = "VERY_LIKELY"
}

export enum GoogleCloudVisionV1p1beta1FaceAnnotationHeadwearLikelihoodEnum {
    Unknown = "UNKNOWN",
    VeryUnlikely = "VERY_UNLIKELY",
    Unlikely = "UNLIKELY",
    Possible = "POSSIBLE",
    Likely = "LIKELY",
    VeryLikely = "VERY_LIKELY"
}

export enum GoogleCloudVisionV1p1beta1FaceAnnotationJoyLikelihoodEnum {
    Unknown = "UNKNOWN",
    VeryUnlikely = "VERY_UNLIKELY",
    Unlikely = "UNLIKELY",
    Possible = "POSSIBLE",
    Likely = "LIKELY",
    VeryLikely = "VERY_LIKELY"
}

export enum GoogleCloudVisionV1p1beta1FaceAnnotationSorrowLikelihoodEnum {
    Unknown = "UNKNOWN",
    VeryUnlikely = "VERY_UNLIKELY",
    Unlikely = "UNLIKELY",
    Possible = "POSSIBLE",
    Likely = "LIKELY",
    VeryLikely = "VERY_LIKELY"
}

export enum GoogleCloudVisionV1p1beta1FaceAnnotationSurpriseLikelihoodEnum {
    Unknown = "UNKNOWN",
    VeryUnlikely = "VERY_UNLIKELY",
    Unlikely = "UNLIKELY",
    Possible = "POSSIBLE",
    Likely = "LIKELY",
    VeryLikely = "VERY_LIKELY"
}

export enum GoogleCloudVisionV1p1beta1FaceAnnotationUnderExposedLikelihoodEnum {
    Unknown = "UNKNOWN",
    VeryUnlikely = "VERY_UNLIKELY",
    Unlikely = "UNLIKELY",
    Possible = "POSSIBLE",
    Likely = "LIKELY",
    VeryLikely = "VERY_LIKELY"
}


// GoogleCloudVisionV1p1beta1FaceAnnotation
/** 
 * A face annotation object contains the results of face detection.
**/
export class GoogleCloudVisionV1p1beta1FaceAnnotation extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=angerLikelihood" })
  angerLikelihood?: GoogleCloudVisionV1p1beta1FaceAnnotationAngerLikelihoodEnum;

  @SpeakeasyMetadata({ data: "json, name=blurredLikelihood" })
  blurredLikelihood?: GoogleCloudVisionV1p1beta1FaceAnnotationBlurredLikelihoodEnum;

  @SpeakeasyMetadata({ data: "json, name=boundingPoly" })
  boundingPoly?: GoogleCloudVisionV1p1beta1BoundingPoly;

  @SpeakeasyMetadata({ data: "json, name=detectionConfidence" })
  detectionConfidence?: number;

  @SpeakeasyMetadata({ data: "json, name=fdBoundingPoly" })
  fdBoundingPoly?: GoogleCloudVisionV1p1beta1BoundingPoly;

  @SpeakeasyMetadata({ data: "json, name=headwearLikelihood" })
  headwearLikelihood?: GoogleCloudVisionV1p1beta1FaceAnnotationHeadwearLikelihoodEnum;

  @SpeakeasyMetadata({ data: "json, name=joyLikelihood" })
  joyLikelihood?: GoogleCloudVisionV1p1beta1FaceAnnotationJoyLikelihoodEnum;

  @SpeakeasyMetadata({ data: "json, name=landmarkingConfidence" })
  landmarkingConfidence?: number;

  @SpeakeasyMetadata({ data: "json, name=landmarks", elemType: GoogleCloudVisionV1p1beta1FaceAnnotationLandmark })
  landmarks?: GoogleCloudVisionV1p1beta1FaceAnnotationLandmark[];

  @SpeakeasyMetadata({ data: "json, name=panAngle" })
  panAngle?: number;

  @SpeakeasyMetadata({ data: "json, name=rollAngle" })
  rollAngle?: number;

  @SpeakeasyMetadata({ data: "json, name=sorrowLikelihood" })
  sorrowLikelihood?: GoogleCloudVisionV1p1beta1FaceAnnotationSorrowLikelihoodEnum;

  @SpeakeasyMetadata({ data: "json, name=surpriseLikelihood" })
  surpriseLikelihood?: GoogleCloudVisionV1p1beta1FaceAnnotationSurpriseLikelihoodEnum;

  @SpeakeasyMetadata({ data: "json, name=tiltAngle" })
  tiltAngle?: number;

  @SpeakeasyMetadata({ data: "json, name=underExposedLikelihood" })
  underExposedLikelihood?: GoogleCloudVisionV1p1beta1FaceAnnotationUnderExposedLikelihoodEnum;
}
